filters/auth/yamlconfig.go

```diff
@@ -0,0 +1,84 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+)
+
+// yamlConfigParser parses and caches yaml configurations of type T.
+// Use [newYamlConfigParser] to create instances and ensure that *T implements [yamlConfig].
+type yamlConfigParser[T any] struct {
+	initialize func(*T) error
```

Comment:
why do we need to enforce this?

Reply:
The idea is to ensure that user of yamlConfigParser is aware that config values must be immutable.